lib/server.js

```diff
@@ -323,10 +334,17 @@ CoAPServer.prototype._handle = function(packet, rsinfo) {
       })
     }
 
-    lru.set(toKey(rsinfo.address, rsinfo.port,
-                  packet, packet.ack || !packet.confirmable), buf)
     buf.sender = sender
 
+    var key = toKey(rsinfo.address, rsinfo.port,
+        packet, packet.ack || !packet.confirmable)
+    lru.set(key, buf)
+
+    // Start timer to actively prune from LRU cache after expiration
+    buf.lru_timer = setTimeout(function () {
+      lru.get(key);
+    }, that._lruMaxAge + 1000)
```

Comment:
you can create an interval and call `unref()`, then you won't need to reset the timer every time.